internal/wasmldr/src/loader/compiled/tls.rs

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::any::Any;
+use std::io::{ErrorKind, IoSlice, IoSliceMut, Read, Write};
+use std::mem::forget;
+use std::ops::{Deref, DerefMut};
+use std::sync::{Arc, RwLock};
+
+use cap_std::net::{TcpListener as CapListener, TcpStream as CapStream};
```

Comment:
This could have just been imported as `TcpListener`(it's the "actual" listener we use) if we avoid importing `wasmtime_wasi::net` primitives at top level. Same for stream